aws_lambda_powertools/utilities/parameters/ssm.py

```diff
@@ -0,0 +1,212 @@
+"""
+AWS SSM Parameter retrieval and caching utility
+"""
+
+
+from typing import Dict, Optional, Union
+
+import boto3
+from botocore.config import Config
+
+from .base import DEFAULT_PROVIDERS, BaseProvider
+
+
+class SSMProvider(BaseProvider):
+    """
+    AWS Systems Manager Parameter Store Provider
+
+    Parameters
+    ----------
+    config: botocore.config.Config, optional
+        Botocore configuration to pass during client initialization
+
+    Example
+    -------
+    **Retrieves a parameter value from Systems Manager Parameter Store**
+
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>> ssm_provider = SSMProvider()
+        >>>
+        >>> ssm_provider.get("/my/parameter")
+
+    **Retrieves a parameter value from Systems Manager Parameter Store in another AWS region**
+
+        >>> from botocore.config import Config
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>>
+        >>> config = Config(region_name="us-west-1")
+        >>> ssm_provider = SSMProvider(config=config)
+        >>>
+        >>> ssm_provider.get("/my/parameter")
+
+    **Retrieves multiple parameter values from Systems Manager Parameter Store using a path prefix**
```

Comment:
Could you add an example of how to access the parameters retrieved? same as the suggestion on Dynamo